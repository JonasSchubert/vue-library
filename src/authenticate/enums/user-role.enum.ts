/* eslint-disable */

export enum UserRole {
  Null = 0,
  User = 1 << 0,
  Administrator = 1 << 1
}

export namespace UserRole {
  export const toArray = (userRole: UserRole): UserRole[] => {
    const array: UserRole[] = [];

    let index = 0;
    let compare: number;

    while (UserRole[compare = 1 << index++]) {
      if (userRole & compare) {
        array.push(compare);
      }
    }

    return array;
  }

  export const toValue = (userRoles: UserRole[]): UserRole => {
    let userRole: UserRole = UserRole.Null;

    userRoles.forEach((value: UserRole) => userRole = userRole + value);

    return userRole;
  }
}
