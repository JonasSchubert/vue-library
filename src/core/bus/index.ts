import { CommunicationBus } from "./type";

export const createCommunicationBus = (): CommunicationBus => {
  const subscribers: { [key: string]: Function[] } = {};

  const unsubscribe = (event: string): void => { subscribers[event].splice(subscribers[event].length - 1, 1); }

  const subscribe = (event: string, callback: Function): Function => {
    if (!Array.isArray(subscribers[event])) {
      subscribers[event] = [];
    }

    subscribers[event].push(callback);

    return (): void => unsubscribe(event)
  };

  const publish = (event: string, data: any): void => {
    if (Array.isArray(subscribers[event])) {
      subscribers[event].forEach((callback: Function) => callback(data));
    }
  };

  return {
    publish,
    subscribe,
    unsubscribe
  }
};
