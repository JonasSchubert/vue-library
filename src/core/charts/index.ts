import BarChart from './bar-chart/bar-chart';
import BubbleChart from './bubble-chart/bubble-chart';
import DoughnutChart from './doughnut-chart/doughnut-chart';
import HorizontalBarChart from './horizontal-bar-chart/horizontal-bar-chart';
import LineChart from './line-chart/line-chart';
import PieChart from './pie-chart/pie-chart';
import PolarChart from './polar-chart/polar-chart';
import RadarChart from './radar-chart/radar-chart';
import ScatterChart from './scatter-chart/scatter-chart';

export const charts = [BarChart, BubbleChart, DoughnutChart, HorizontalBarChart, LineChart, PieChart, PolarChart, RadarChart, ScatterChart];
