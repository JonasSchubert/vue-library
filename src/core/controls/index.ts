import AppBarBtn from './app-bar-btn/app-bar-btn.vue';
import AppSnackbar from './app-snackbar/app-snackbar.vue';

export const controls = [AppBarBtn, AppSnackbar];
