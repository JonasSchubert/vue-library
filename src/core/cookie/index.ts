export const deleteCookie = (key: string): void => {
  document.cookie = `${key}=;expires=Thu, 01 Jan 1970 00:00:00 UTC`;
};

export const readCookie = (key: string): string | undefined => {
  let data;

  const documentCookieData: string[] = document.cookie.split(";");
  for (let index = 0; index < documentCookieData.length; index += 1) {
    const value: string = documentCookieData[index];
    if (value.includes(`${key}=`)) {
      data = value.replace(`${key}=`, "").trim();
      break;
    }
  }

  return data;
};

export const writeCookie = (key: string, value: string, daysTilExpired = 1): void => {
  const date: Date = new Date();
  date.setTime(date.getTime() + (daysTilExpired * 24 * 60 * 60 * 1000));
  document.cookie = `${key}=${value};expires=${date.toUTCString()}`;
};
