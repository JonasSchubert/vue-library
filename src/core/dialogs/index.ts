import ConfirmDialog from './confirm-dialog/confirm-dialog.vue';

export const dialogs = [ConfirmDialog];
