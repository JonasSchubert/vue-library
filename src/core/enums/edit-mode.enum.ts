// eslint-disable-next-line no-shadow
export enum EditMode {
  Null = 0,
  Add = 1,
  Update = 2,
  Delete = 3
}
