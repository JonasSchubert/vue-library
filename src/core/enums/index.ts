export * from './edit-mode.enum';
