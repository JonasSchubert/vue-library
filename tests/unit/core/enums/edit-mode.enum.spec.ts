import { EditMode } from '@/core/enums';

describe('EditMode', () => {
  test('should not change unexpected', () => {
    // Assert
    expect(EditMode).toMatchSnapshot();
  });
});
