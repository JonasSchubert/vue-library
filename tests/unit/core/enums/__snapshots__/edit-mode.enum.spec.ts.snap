// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMode should not change unexpected 1`] = `
Object {
  "0": "Null",
  "1": "Add",
  "2": "Update",
  "3": "Delete",
  "Add": 1,
  "Delete": 3,
  "Null": 0,
  "Update": 2,
}
`;
